src/rules/no-invalid-css.ts

```diff
@@ -0,0 +1,60 @@
+/**
+ * @fileoverview Disallows invalid CSS in templates
+ * @author Kristján Oddsson <https://github.com/koddsson>
+ */
+
+import {Rule} from 'eslint';
+import * as ESTree from 'estree';
+import {validate as validateCSS} from 'csstree-validator';
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+const rule: Rule.RuleModule = {
+  meta: {
+    docs: {
+      description: 'Disallows invalid CSS in templates',
+      recommended: false,
+      url: 'https://github.com/43081j/eslint-plugin-lit/blob/master/docs/rules/no-invalid-css.md'
+    },
+    schema: [],
+    messages: {
+      parseError: 'Template contained invalid CSS syntax, error was: {{ err }}'
+    }
+  },
+
+  create(context): Rule.RuleListener {
+    //----------------------------------------------------------------------
+    // Helpers
+    //----------------------------------------------------------------------
+
+    //----------------------------------------------------------------------
+    // Public
+    //----------------------------------------------------------------------
+
+    return {
+      TaggedTemplateExpression: (node: ESTree.Node): void => {
+        if (
+          node.type === 'TaggedTemplateExpression' &&
+          node.tag.type === 'Identifier' &&
+          node.tag.name === 'css'
+        ) {
+          const errors = validateCSS(
+            node.quasi.quasis.map((quasi) => quasi.value.raw).join('')
```

Comment:
similar to why i'm not hugely active maintaining postcss-lit - this stuff becomes very finicky

think about:

```ts
css`
  .foo {
    color: ${someColor};
  }
`;
```

this'll be validated as:

```css
.foo {
  color: ;
}
```

which presumably fails validation (if it doesn't, there'll be similar cases which do)

in postcss-lit we just tried our best to choose a sensible placeholder, but there's still endless edge cases 😞 

maybe thats what we could do here though as "best effort"?

you can see what we do here:
https://github.com/43081j/postcss-lit/blob/b19673062b3f0a63f4be3eaa1845018d1460f156/src/util.ts#L81-L100

but maybe we can be looser than that and just tell consumers to turn it off if it breaks

Reply:
Yeah it's a huge PITA. I was thinking that one could turn on this rule if they wanted to strictly not allow template interpolation like this. Or it could emit a warning if plugins like `rollup-plugin-minify-html-templates` won't be able to minify the given template.

https://github.com/43081j/postcss-lit/blob/b19673062b3f0a63f4be3eaa1845018d1460f156/src/util.ts#L125-L191

is something that's needed in `rollup-plugin-minify-html-templates!

Maybe I'll put this back into draft mode so I can document this better and maybe work around _some_ of the edge cases as a best effort.